src/test/java/com/google/step/youtube/YoutubeInfoScraperTest.java

```diff
@@ -14,132 +14,209 @@
+    }
+
+    /** Multiple items with one item having no promo-code in the description. */
+    @Test
+    public void scrapePromoCodesFromPlaylist_multipleItemsSomeCodesFound() throws IOException {
+        mockPlaylistResponse = new PlaylistItemListResponse();
+        String description1 = "Get 20% off your first monthly box when you sign up at "
+                + "http://boxofawesome.com and enter the code RO at checkout!";
+        String description2 = "Use code LINUS and get 25% off GlassWire at https://lmg.gg/glasswire";
+        String descriptionWithNoCode = "Check out our UPDATED version which has all the NEW ELEMENTS here: "
+                + "\n https://youtu.be/rz4Dd1I_fX0  The TEETH Song (Memorize Every Tooth): https://youtu.be/PI3hne8C8rU"
+                + "\nDownload on ITUNES: http://bit.ly/12AeW99 ";
+        mockPlaylistResponse.setItems(Arrays.asList(
+                new PlaylistItem().setSnippet(new PlaylistItemSnippet().setDescription(description1)
+                        .setPublishedAt(new DateTime(MOCK_DATE)).setResourceId(new ResourceId().setVideoId(VIDEO_ID))),
+                new PlaylistItem().setSnippet(new PlaylistItemSnippet().setDescription(description2)
+                        .setPublishedAt(new DateTime(MOCK_DATE)).setResourceId(new ResourceId().setVideoId(VIDEO_ID))),
+                new PlaylistItem().setSnippet(new PlaylistItemSnippet().setDescription(descriptionWithNoCode)
+                        .setPublishedAt(new DateTime(MOCK_DATE))
+                        .setResourceId(new ResourceId().setVideoId(VIDEO_ID)))));
+        when(mockListPlaylistItems.execute()).thenReturn(mockPlaylistResponse);
+        Optional<List<PromoCode>> actual = scraper.scrapePromoCodesFromPlaylist(UPLOAD_ID);
+        assertThat(actual.get(), equalTo(Arrays.asList(PromoCode.create("RO", VIDEO_ID, new Date(MOCK_DATE)),
+                PromoCode.create("LINUS", VIDEO_ID, new Date(MOCK_DATE)))));
+    }
+
+    @Test
+    public void scrapePromoCodesFromPlaylist_oneItemNoCodeFound() throws IOException {
+        mockPlaylistResponse = new PlaylistItemListResponse();
+        String descriptionWithNoCode = "Check out our UPDATED version which has all the NEW ELEMENTS here: "
+                + "\n https://youtu.be/rz4Dd1I_fX0  The TEETH Song (Memorize Every Tooth): https://youtu.be/PI3hne8C8rU"
+                + "\nDownload on ITUNES: http://bit.ly/12AeW99 ";
+        mockPlaylistResponse.setItems(Arrays
+                .asList(new PlaylistItem().setSnippet(new PlaylistItemSnippet().setDescription(descriptionWithNoCode)
+                        .setPublishedAt(new DateTime(0L)).setResourceId(new ResourceId().setVideoId(VIDEO_ID)))));
+        when(mockListPlaylistItems.execute()).thenReturn(mockPlaylistResponse);
+        Optional<List<PromoCode>> actual = scraper.scrapePromoCodesFromPlaylist(UPLOAD_ID);
+        assertThat(actual.get(), equalTo(Arrays.asList()));
```

Comment:
checking on isEmpty() is more readable here.

Reply:
Done.